packages/meditrak-server/src/routes/importSurveyResponses/SurveyResponseUpdatePersistor.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+const RECORDS_PER_BULK_BATCH = 5000; // number of records (survey responses + answers) processed per bulk insert/delete
+
+export const CREATE = 'create';
+export const UPDATE = 'update';
+export const DELETE = 'delete';
+
+function batchCreates(creates) {
```

Comment:
I didn't thoroughly review this function - looks good at a first glance!

Reply:
I did, but I don't consistently make an active decision about this. Basically, if I bother thinking about it, my decision tree is:

1. Are there already functions in this area that use one or the other? If so, match them. If not, move to 2.
2. Will it be used as a variable and passed around, e.g. as a callback etc.? If so, declare it as a variable using an arrow function. If not, move to 3.
3. If you get here, use a regular old function

The decision between arrow and regular functions resting on whether it's treated as a variable comes from @mcccclean's handbook of conventions (that should be a real publication btw McLean)

Reply:
That's interesting! I personally like the elegance of arrow functions so I use them everywhere (even for class methods). To my knowledge, the only real difference is management of `this`, and I don't find myself manipulating `this` in a non-class function ever.

In fact, handling of `this` in regular (non-arrow) class methods can be tricky. Consider the following example:

```js
class Test {
  constructor() {
    this.x = 0;
  }

  incrementRegular(a) {
    this.x += a;
  }

  incrementArrow = a => {
    this.x += a;
  };
}

arrowTest = new Test();
[2, 3].forEach(arrowTest.incrementArrow);
console.log('arrow', arrowTest.x) // 5

regularTest = new Test();
[2, 3].forEach(regularTest.incrementRegular);
console.log('regular', regularTest.x); ; // Will throw `TypeError: Cannot read property 'x' of undefined`

```

The above snippet will throw an error, because `this` will be `undefined`. in the `forEach(regularTest.incrementRegular)` call.

That being said, while arrow functions do work for classes, they don't for objects:

```js
const regularObject = {
  x: 1,
  getXRegular: function () {
    return this.x;
  },
  getXArrow: () => this.x,
};

console.log(regularObject.getXRegular()); // 1
console.log(regularObject.getXArrow()); // undefined
```

I do agree with point 1 though - if there is an existing style, better to stick to it... or refactor everything else 😆